part_1/solution.py

```diff
@@ -0,0 +1,1327 @@
+specified by the Open Microscopy Environment Next Generation File Format
+(OME-NGFF).
+
+- The layout on the disk is: `row/col/field/pyramid_level/timepoint/channel/z/y/x.`
+- These datasets only have 1 level in the pyramid (highest resolution) which is '0'.
+"""
+
+# %% [markdown]
+"""
+<div class="alert alert-warning">
+You can inspect the tree structure by using your terminal:
+<code> iohub info -v "path-to-ome-zarr" </code>
+
+<br>
+More info on the CLI:
+<code>iohub info --help </code> to see the help menu.
+</div>
+"""
+# %%
+# This is the python function called by `iohub info` CLI command
+print_info(data_path, verbose=True)
+
+# Open and inspect the dataset.
+dataset = open_ome_zarr(data_path)
+
+# %%
+# Use the field and pyramid_level below to visualize data.
+row = 0
+col = 0
+field = 9  # TODO: Change this to explore data.
+
+
+# NOTE: this dataset only has one level
+pyaramid_level = 0
+
+# `channel_names` is the metadata that is stored with data according to the OME-NGFF spec.
+n_channels = len(dataset.channel_names)
+
+image = dataset[f"{row}/{col}/{field}/{pyaramid_level}"].numpy()
+print(f"data shape: {image.shape}, FOV: {field}, pyramid level: {pyaramid_level}")
+
+figure, axes = plt.subplots(1, n_channels, figsize=(9, 3))
+
+for i in range(n_channels):
+    for i in range(n_channels):
+        channel_image = image[0, i, 0]
+        # Adjust contrast to 0.5th and 99.5th percentile of pixel values.
+        p_low, p_high = np.percentile(channel_image, (0.5, 99.5))
+        channel_image = np.clip(channel_image, p_low, p_high)
+        axes[i].imshow(channel_image, cmap="gray")
+        axes[i].axis("off")
+        axes[i].set_title(dataset.channel_names[i])
+plt.tight_layout()
+
+# %% [markdown]
+# <div class="alert alert-info">
+#
+# ### Task 1.1
+#
+# Look at a couple different fields of view by changing the value in the cell above.
+# Check the cell density, the cell morphologies, and fluorescence signal.
+# </div>
+
+# %% [markdown]
+"""
+## Explore the effects of augmentation on batch.
+
+VisCy builds on top of PyTorch Lightning. PyTorch Lightning is a thin wrapper around PyTorch that allows rapid experimentation. It provides a [DataModule](https://lightning.ai/docs/pytorch/stable/data/datamodule.html) to handle loading and processing of data during training. VisCy provides a child class, `HCSDataModule` to make it intuitve to access data stored in the HCS layout.
+  
+The dataloader in `HCSDataModule` returns a batch of samples. A `batch` is a list of dictionaries. The length of the list is equal to the batch size. Each dictionary consists of following key-value pairs.
+- `source`: the input image, a tensor of size 1*1*Y*X
+- `target`: the target image, a tensor of size 2*1*Y*X
+- `index` : the tuple of (location of field in HCS layout, time, and z-slice) of the sample.
+"""
+
+# %% [markdown]
+# <div class="alert alert-info">
+#
+# ### Task 1.2
```

Comment:
Is there anything for them to do aside from run the code? Couldn’t find differences between exercise and solution notebook
    - If yes… add explicit TODO 
    - If no… maybe add questions about what they should look for/think about